packages/flutter/lib/src/widgets/widget_span.dart

```diff
@@ -0,0 +1,262 @@
+        if (!child.visitChildren(visitor))
+          return false;
+      }
+    }
+    return true;
+  }
+
+  String toPlainText() {
+    return widget.toString();
+  }
+
+  int codeUnitAt(int index) {
+    return null;
+  }
+
+  /// Describe the difference between this widget span and another [TextSpan],
+  /// in terms of how much damage it will make to the rendering. The comparison
+  /// is deep.
+  ///
+  /// Comparing a [WidgetSpan] with a [TextSpan] will result in [RenderComparison.layout].
+  ///
+  /// See also:
+  ///
+  ///  * [TextStyle.compareTo], which does the same thing for [TextStyle]s.
+  RenderComparison compareTo(InlineSpan other) {
+    if (identical(this, other))
+      return RenderComparison.identical;
+    if (other.runtimeType is! WidgetSpan)
+      return RenderComparison.layout;
+    final WidgetSpan typedOther = other;
+    if (typedOther.widget != widget)
+      return RenderComparison.layout;
+    if ((style == null) != (other.style == null))
+      return RenderComparison.layout;
+    RenderComparison result = RenderComparison.identical;
+    if (style != null) {
+      final RenderComparison candidate = style.compareTo(other.style);
+      if (candidate.index > result.index)
+        result = candidate;
+      if (result == RenderComparison.layout)
+        return result;
+    }
+    // WidgetSpans are always a leaf node in the TextSpan tree, and have
+    // no children.
+    return result;
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other.runtimeType != runtimeType)
+      return false;
+    final WidgetSpan typedOther = other;
+    return typedOther.widget == widget
+        && typedOther.style == style;
+  }
+
+  /// Returns the text span that contains the given position in the text.
+  TextSpan getSpanForPosition(TextPosition position) {
+    assert(debugAssertIsValid());
+    return null;
+  }
+
+
+  /// In checked mode, throws an exception if the object is not in a
+  /// valid configuration. Otherwise, returns true.
+  ///
+  /// This is intended to be used as follows:
+  ///
+  /// ```dart
+  /// assert(myTextSpan.debugAssertIsValid());
+  /// ```
+  bool debugAssertIsValid() {
+    assert(() {
+      if (!visitChildren((InlineSpan span) {
+        if (span is WidgetSpan)
+          return (span as WidgetSpan).widget != null;
+        TextSpan textSpan = span;
+        if (textSpan.children != null) {
+          for (InlineSpan child in textSpan.children) {
+            if (child == null)
+              return false;
+          }
+        }
+        return true;
+      })) {
+        throw FlutterError(
+          'TextSpan contains a null child.\n'
+          'A TextSpan object with a non-null child list should not have any nulls in its child list.\n'
+          'The full text in question was:\n'
+          '${toStringDeep(prefixLineOne: '  ')}'
+        );
+      }
+      return true;
+    }());
+    return true;
+  }
+}
```

Comment:
I think this whole method can be a no-op. The only thing you actually have to test is `child != null` and you can do that in the constructor.